src/Future.ts

```diff
@@ -0,0 +1,62 @@
+import * as Async from "./Async";
+import * as Fx from "./Fx";
+
+export interface Future<E, A> {
+  readonly state: FutureState<E, A>;
+  readonly wait: Fx.Fx<Async.Async | E, A>;
+  readonly complete: (fx: Fx.Fx<E, A>) => boolean;
```

Comment:
What's your thinking on completing with an Fx of A rather than just A?

Reply:
Familiarity mostly, it's the way in which Effect's `Deferred` datatype works as well. On the practical side, it allows propagating requirements, like failures, to other Fibers